### ISSUE TITLE
Train on full GQA dataset

### ISSUE THREAD

**Issue by kehanlu** [stanfordnlp/mac-network#26]
hi, I currently work on this dataset.
I followed the guide and successfully train on Data1.2.zip and CLEVR version.
but now I want to train on the full Dataset from the website (70G).

1. Is Dataset1.2.zip a small subset of which on the website, or it has all questions and images (just remove some unnecessary part)?
2. if yes, then what should I do to run on "full" dataset with the baseline model?

thanks for your great work

**Comment by dorarad** [comment]
Hi, thanks a lot for the interest in the dataset! 
dataset1.2 contains all the questions. The reason why the dataset on the website is bigger is mainly the image features, which we handle in the repository in the "Feature extraction" section (they are in a different zip file outside of dataset1.2).

However, the default settings make the model run on the "balanced" training set, which is a subset of the whole training data (1M out of 14M training samples) with reduced biases in questions etc. Regardless of what the models are being trained on, the test (in the challenge etc) is performed only on the balanced test set (so that models can't cheat by exploiting question biases). In experiments I've done I saw that baselines get no performance improvement on the balanced test set when training on the full training set (14M) instead of the balanced training set (1M).

However, if you would like to train on the full dataset, all that is needed to do is to pass the following flag: `--dataSubset all`. 
(note: in that case epoch time will get to be about 10-12hours as each epoch will run over 14M questions).

Hope it helps! Please let me know if you have any other questions!

**Comment by kehanlu** [comment]
I think I get the point, will keep trying
thanks a lot :)

**Comment by dorarad** [comment]
happy to help! :)
